service-commands/scripts/bake-dev-image.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# bakes from latest master on gcp by default
+
+set -x
+
+# error handling
+trap handle_provisioning_error 1
```

Comment:
For my own knowledge, how does `trap` work again?

Reply:
tldr allows you to catch exit signals and run code when they occur! [here's more](https://www.linuxjournal.com/content/bash-trap-command) - very useful!

Reply:
Oh! So this is even better than `set -e` since it will catch any `exit 1` instances in this script and then send a prettyprint message?

Reply:
Yep! You can define different trap statements for different exit codes and error handling for those. It's quite useful! :D